crates/sui-indexer-alt-graphql/src/api/types/epoch.rs

```diff
@@ -67,6 +66,20 @@ impl Epoch {
     async fn end(&self, ctx: &Context<'_>) -> Result<EpochEnd, RpcError> {
         EpochEnd::fetch(ctx, &self.start.scope, self.epoch_id).await
     }
+
+    /// The total number of checkpoints in this epoch.
+    async fn total_checkpoints(&self, ctx: &Context<'_>) -> Result<Option<UInt53>, RpcError> {
+        let last = match &self.end(ctx).await?.contents {
+            Some(last) => last.cp_hi as u64,
+            None => self.start.scope.checkpoint_viewed_at(),
+        };
+        let first = match &self.start.contents {
```

Comment:
This doesn't quite work -- to see why, add a test for:

```graphql
{
  checkpoint {
    epoch {
      totalCheckpoints
    }
}
```

This will return `None` because we did not eagerly fetch `start` when fetching the checkpoint's epoch. The longer suggestion above will fix that.